src/middleware/auth.ts

```diff
@@ -65,5 +65,31 @@ async function adminAuthImpl(req: Request, res: Response, next: NextFunction) {
   }
 }
 
+async function passwordResetAuthImpl(req: Request, res: Response, next: NextFunction) {
+  try {
+    const token = req.header('x-auth-token');
+
+    if (!token) throw new UnauthorizedError('No password reset token found!');
+
+    const decoded = jwt.decode(token) as UserDecoded;
```

Comment:
Can decode without `secretKey`?

Reply:
Probably need a `RESET_TOKEN_SECRET` env variable for the secret token

Reply:
Yea. Can be decoded. I tested using postman too

Reply:
👍